Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m

```diff
@@ -0,0 +1,688 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <objc/runtime.h>
+#include <sys/utsname.h>
+
+#import <GoogleDataTransport/GDTEvent.h>
+#import <GoogleDataTransport/GDTTargets.h>
+#import <GoogleDataTransport/GDTTransport.h>
+
+#import <GoogleUtilities/GULAppEnvironmentUtil.h>
+#import <GoogleUtilities/GULLogger.h>
+
+#import <FirebaseCoreDiagnosticsInterop/FIRCoreDiagnosticsData.h>
+#import <FirebaseCoreDiagnosticsInterop/FIRCoreDiagnosticsInterop.h>
+
+#import <nanopb/pb.h>
+#import <nanopb/pb_decode.h>
+#import <nanopb/pb_encode.h>
+
+#import "FIRCDLibrary/Protogen/nanopb/firebasecore.nanopb.h"
+
+/** The logger service string to use when printing to the console. */
+static GULLoggerService kFIRCoreDiagnostics = @"[FirebaseCoreDiagnostics/FIRCoreDiagnostics]";
+
+#define STR(x) STR_EXPAND(x)
+#define STR_EXPAND(x) #x
+
+#ifdef FIREBASE_BUILD_ZIP_FILE
```

Comment:
What's the process to build differently for CocoaPods, Carthage, and zip distro?

Reply:
For Cocoapods, since the ruby logic doesn't seem to carry over to json, it'll be listed as a hard-dep. For Carthage and zip, FirebaseCoreDiagnostics should just be built as another framework (however that's done). If additional stuff is needed, I'll probably need some direction setting up Carthage and zip file builders to build these libraries (GDT, GDTCCT, FirebaseCoreDiagnosticsInterop, and FirebaseCoreDiagnostics)

Reply:
Building in non-Cocoapod build systems will be resolved in another PR.